From 8c3f9ebfc35ccc288bd492c8d88e2e06fbc8daf6 Mon Sep 17 00:00:00 2001
From: Alex Leibovich <alexl@marvell.com>
Date: Thu, 3 Jan 2019 12:05:51 +0200
Subject: [PATCH 0834/1051] arm64: dts: marvell: add current-limiter regulator
 to USB3 in a7k/a8k DB

Before this patch, the current-limiter regulator relied on U-Boot proper
setting for 900mA. this patch removed this dependency by fixing the
regulator settings to properly use 900mA when needed.

- The USB Host should be linked to current-limiter through
"current-limiter-supply" DTS property.
- This patch adds the "regulator-fixed" node for the USB current limiter.
- This patch adds handling for "current-limiter-supply" property inside
the USB3 node in armada-70x0/80x0-db device trees for turning on
the current limiter.

Change-Id: If0e3a9e763296514586f26c162a73850007b4a50
Signed-off-by: Alex Leibovich <alexl@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/62386
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/1564
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 .../boot/dts/marvell/armada-7040-db.dtsi      | 19 +++++++++++++++
 .../boot/dts/marvell/armada-8040-db.dtsi      | 23 +++++++++++++++++++
 2 files changed, 42 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index e1d77cae0ea5..b822c96ac2e1 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -49,12 +49,31 @@
 
 	cp0_usb3_0_phy: cp0-usb3-0-phy {
 		compatible = "usb-nop-xceiv";
+		/* vcc-supply can be changed by io-expander or GPIO,
+		 * for A7040-DB we are using by default io-expander
+		 * for other board GPIO may be needed, should update
+		 * the phandle to gpio regulator
+		 */
 		vcc-supply = <&cp0_reg_usb3_0_vbus>;
+		current-limiter-supply = <&exp_usb3_0_current_limiter>;
 	};
 
 	cp0_usb3_1_phy: cp0-usb3-1-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp0_reg_usb3_1_vbus>;
+		current-limiter-supply = <&exp_usb3_1_current_limiter>;
+	};
+	exp_usb3_0_current_limiter: usb3-0-current-limiter {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-0-current-limiter";
+		enable-active-high;
+		gpio = <&expander0 4 GPIO_ACTIVE_HIGH>;
+	};
+	exp_usb3_1_current_limiter: usb3-1-current-limiter {
+		compatible = "regulator-fixed";
+		regulator-name = "usb3-1-current-limiter";
+		enable-active-high;
+		gpio = <&expander0 5 GPIO_ACTIVE_HIGH>;
 	};
 };
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 883a9afe2255..6722665c279e 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -37,12 +37,28 @@
 	cp0_usb3_0_phy: cp0-usb3-0-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp0_reg_usb3_0_vbus>;
+		current-limiter-supply = <&cp0_reg_usb3_0_current_limiter>;
+	};
+	cp0_reg_usb3_0_current_limiter: cp0_usb3-current-limiter0 {
+		compatible = "regulator-fixed";
+		regulator-name = "cp0-usb3-current-limiter-h0";
+		enable-active-high;
+		gpio = <&expander0 4 GPIO_ACTIVE_HIGH>;
 	};
 
+
 	cp0_usb3_1_phy: cp0-usb3-1-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp0_reg_usb3_1_vbus>;
+		current-limiter-supply = <&cp0_reg_usb3_1_current_limiter>;
 	};
+	cp0_reg_usb3_1_current_limiter: cp0_usb3-current-limiter1 {
+		compatible = "regulator-fixed";
+		regulator-name = "cp0-usb3-current-limiter-h1";
+		enable-active-high;
+		gpio = <&expander0 5 GPIO_ACTIVE_HIGH>;
+	};
+
 
 	cp1_reg_usb3_0_vbus: cp1-usb3-0-vbus {
 		compatible = "regulator-fixed";
@@ -65,8 +81,15 @@
 	cp1_usb3_0_phy: cp1-usb3-0-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp1_reg_usb3_0_vbus>;
+		current-limiter-supply = <&cp1_reg_usb3_0_current_limiter>;
 	};
 
+	cp1_reg_usb3_0_current_limiter: cps_usb3-current-limiter0 {
+		compatible = "regulator-fixed";
+		regulator-name = "cps-usb3-current-limiter-h0";
+		enable-active-high;
+		gpio = <&expander1 4 GPIO_ACTIVE_HIGH>;
+	};
 	cp1_usb3_1_phy: cp1-usb3-1-phy {
 		compatible = "usb-nop-xceiv";
 		vcc-supply = <&cp1_reg_usb3_1_vbus>;
-- 
2.17.1

